spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -225,6 +241,12 @@ public Scan build() {
       scan = scan.asOfTime(asOfTimestamp);
     }
 
+    if (branch != null) {
+      scan.useSnapshotRef(branch);
+    } else if (tag != null) {
```

Comment:
I think this should not use `else`. Just set the ref and let the `TableScan` handle ensuring there is only one call.